superbench/benchmarks/micro_benchmarks/micro_base.py

```diff
@@ -21,8 +22,10 @@ def __init__(self, name, parameters=''):
         """
         super().__init__(name, parameters)
         self._benchmark_type = BenchmarkType.MICRO
+        # Set the binary directory.
+        self._binary_dir = os.getenv('SUPERBENCH_BINARY_DIR', '/usr/local/bin')
```

Comment:
what is this used for? `subprocess.run` can execute executables in system path directly. finding the path is what os does

Reply:
SUPERBENCH_BINARY_DIR is the environment variable for superbench binary dir, like /opt/superbench/bin, all microbenchmarks will find the binary from the directory.

Reply:
yes those binaries will be placed in one directory, but I mean we can find/run them directly without the prefix
besides, you may run through mpirun/python which won't be placed in superbench directory

Reply:
Didn't get your point. Just want to confirm we are on the same page:

**Dependency (binary and library)**
As we discussed, we will handle the third-party projects and ourselves projects in two ways, third-party projects will be put as submodules in superbenchmark/thirdparty, and ourselves project will be put at superbenchmark/superbench/benchmarks/micro_benchmarks/xxx/. Both of them will be constructed using cmake or make, and the the binary will be put at SUPERBENCH_BINARY_DIR, for example /opt/superbench/bin, and library will be put at SUPERBENCH_LIBRARY_DIR, for example /opt/superbench/lib. For this part, I thought we have reach an agreement from last discussion, right?

**Micro-Benchmarks**
All our benchmarks are wrote in python and inherit from the base class "Benchmark", executor only interact with benchmarks through BenchmarkRegistry as we discussed before. Even you run through mpirun, the command executed by mpirun is our python benchmark, not the binary (dependency) itself. I don't think it's a good way for runner or executor itself to have another way to create and maintain some benchmarks. We should manage all benchmarks in a unify way.

We can have more discussion if there have anything unclear. @abuccts @cp5555